scripts/travis_target.sh

```diff
@@ -1,85 +1,75 @@
-#! /bin/bash
+#!/bin/bash
+set -ev
 
-# Travis cpp jobs construct a matrix based on environment variables
-# (and the value of 'compiler'). In order to test multiple builds
-# (release/debug/cpplint/cppcheck), this uses a TRAVIS_TARGET env
-# var to do the right thing.
+# Set compiler to GCC 4.8 here, as Travis overrides the global variables.
+export CC=gcc-4.8 CXX=g++-4.8
 
-function travis_make()
-{
-    mkdir $1 && cd $1
+if [ ${TRAVIS_TARGET} == CPPCHECK ]; then
+  # grab a pre-built cppcheck from s3
+  wget https://s3.amazonaws.com/kylo-pl-bucket/pcre-8.36_install.tar.bz2
+  tar xjvf pcre-8.36_install.tar.bz2 --strip 1 -C $USERDIR
+  wget https://s3.amazonaws.com/kylo-pl-bucket/cppcheck-1.69_install.tar.bz2
+  tar xjvf cppcheck-1.69_install.tar.bz2 --strip 1 -C $USERDIR
+elif [ ${TRAVIS_TARGET} == DOXYGEN ]; then
+  # grab a pre-built doxygen 1.8.7 from s3
+  wget https://s3.amazonaws.com/kylo-pl-bucket/doxygen_install.tar.bz2
+  tar xjvf doxygen_install.tar.bz2 --strip 1 -C $USERDIR
+elif [ ${TRAVIS_TARGET} == DEBUG ]; then
+  # Install coveralls.io update utility
+  pip install --user cpp-coveralls
+fi
 
-    # Set compiler to GCC 4.8 here, as Travis overrides the global variables.
-    export CC=gcc-4.8 CXX=g++-4.8
-    # Generate build files
-    [ $1 == "debug" ] && export CMAKE_VARS="-DCMAKE_BUILD_TYPE=Debug -DCOVERALLS=ON"
-    cmake $CMAKE_VARS ..
-    if [ $? -ne 0 ]; then
-        echo "cmake failed."
-        exit 1
+# Generate build files
+if [ ${TRAVIS_TARGET} == COMMITS ]; then
+  shopt -s nocasematch
+  git log --no-merges --pretty=%s master..$HEAD | while read line ; do
+    if [[ ! "$line" =~ ^\(maint|doc|packaging|pup-\d+\) ]]; then
+      echo -e \
+          "\n\n\n\tThis commit summary didn't match CONTRIBUTING.md guidelines:\n" \
+          "\n\t\t$line\n" \
+          "\tThe commit summary (i.e. the first line of the commit message) should start with one of:\n" \
+          "\t\t(FACT-<digits>) # this is most commond and should be a ticket at tickets.puppetlabs.com\n" \
+          "\t\t(doc)\n" \
+          "\t\t(maint)\n" \
+          "\t\t(packaging)\n" \
+          "\n\tThis test for the commit summary is case-insensitive.\n\n\n"
+      exit 1
     fi
+  done
+  exit 0
+elif [ ${TRAVIS_TARGET} == DEBUG ]; then
```

Comment:
Shouldn't this be RELEASE?  The release build in Travis built debug with these changes.